In [4]:
import os
from pathlib import Path
import numpy as np
import pandas as pd
import datetime as dt

In [2]:
from google.colab import drive
drive.mount('/content/gdrive', force_remount=True)

Mounted at /content/gdrive


In [7]:
# file_path = Path(__file__).parent.joinpath('Files', 'TSLA_quote.csv')
file_path = '/content/gdrive/MyDrive/Colab Notebooks/Files/TSLA_quote.csv'
df_tesla = pd.read_csv(file_path)
df_tesla['Date'] = df_tesla['Date'].apply(pd.to_datetime)
df_tesla.head()

Date Close/Last     Volume      Open       High        Low
0 2024-11-07    $296.91  117309200   $288.89    $299.75    $285.52
1 2024-11-06    $288.53  165228700   $284.67    $289.59    $275.62
2 2024-11-05    $251.44   69282510   $247.34  $255.2799  $246.2101
3 2024-11-04    $242.84   68802350   $244.56    $248.90    $238.88
4 2024-11-01    $248.98   57544760  $252.043    $254.00    $246.63

In [8]:
start_date = dt.datetime(2017,1,20)
end_date = dt.datetime(2023,11,30)

In [9]:
df_tesla.loc[df_tesla.Date==start_date, 'Close/Last']

1963    $16.3153
Name: Close/Last, dtype: object

In [10]:
df_tesla.loc[df_tesla.Date==end_date, 'Close/Last']

236    $240.08
Name: Close/Last, dtype: object

In [11]:
start_price = df_tesla.loc[df_tesla.Date==start_date, 'Close/Last'].item()
start_price = float(start_price.replace('$',''))
print(start_price)

16.3153


In [12]:
end_price = df_tesla.loc[df_tesla.Date==end_date, 'Close/Last'].item()
end_price = float(end_price.replace('$',''))
print(end_price)

240.08


In [13]:
100*(end_price - start_price)/start_price

1371.5022095824165

During my time at Tesla, the stock price increased 1371.5%